docs/blockstates/extendedStates.md

```diff
@@ -0,0 +1,82 @@
+
+*Extended States* and *Unlisted Properties* are Forge-introduced extensions to the blockstates system that address this problem.
+
+!!! Note
+	Extended states and unlisted properties are systems intended to address this problem only in the context of rendering.
+	Although the logical server still has the code and infrastructure for extended states, it cannot communicate it over the network, and nearly all server-side logical actions disregard extended states completely.
+	As a result, extended states are intended to only be used on the logical client.
+
+Unlisted Properties
+-----------------------
+
+Unlisted properties are an alternative kind of `IProperty<V>`, where every possible value combination is *not* generated on startup. Because of this, and because they do not show up on the game's F3 debug menu, we call them "unlisted".
+The interface type for unlisted properties is `IUnlistedProperty<V>`. The interface methods for both types have nearly identical semantics, except that instead of `IProperty.getAllowedValues()`, which returns a collection of all valid values, we have `IUnlistedProperty.isValid(V)` to detect whether a value is valid.
+
+Examples:
+* Forge itself has a [`PropertyFloat`](https://github.com/MinecraftForge/MinecraftForge/blob/e3777f455991b6803eb826ff6305d17749462a3b/src/main/java/net/minecraftforge/common/property/PropertyFloat.java), which can hold any float value passing a certain Predicate.
+* Botania has a generic [`PropertyObject`](https://github.com/Vazkii/Botania/blob/c58ae21173ff63da2f107e328f19a31e0d609778/src/main/java/vazkii/botania/api/state/PropertyObject.java) used to pass any sort of `Object` on to a custom model.
+
+Declaring Unlisted Properties
+------------------------------------
+
+Unlisted properties are declared in `Block.createBlockState`, the same place as regular ("listed") properties. Instead of returning a `BlockStateContainer`, one must return an `ExtendedBlockState`. Forge provides a builder `BlockStateContainer.Builder`, which will automatically handle returning an `ExtendedBlockState` for you.
+
+Example:
+```Java
+@Override
+public BlockStateContainer createBlockState() {
+	return new BlockStateContainer.Builder(this).add(LISTED_PROP).add(UNLISTED_PROP).build();
+}
+```
+
+Filling Extended States
+----------------------------
+
+Before an `IBlockState` is used for rendering purposes, it will always have `Block.getExtendedState` called on it first. In this method, you will give all your unlisted properties values. Assuming you registered at least one unlisted property in the previous section, the `IBlockState` parameter can be safely casted to `IExtendedBlockState`, which has a `withProperty` method for unlisted properties analogous to its listed property cousin. Here, you can query whatever you want from the World, the Tile Entity, etc. (with appropriate safety checks, of course) and insert it into the extended blockstate.
```

Comment:
"Before an `IBlockState` is passed to an `IBakedModel`" would perhaps be a bit more specific.